sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/maskExpressions.scala

```diff
@@ -321,3 +322,37 @@ object Mask {
     }
   }
 }
+
+object MaskExpressionBuilder extends ExpressionBuilder {
+  override def functionSignatures: Seq[FunctionSignature] = {
+    val strArg = NamedArgument("str", FixedArgumentType(StringType))
+    val upperCharArg = NamedArgument(
+      "upperChar",
+      FixedArgumentType(StringType),
+      Some(Literal(Mask.MASKED_UPPERCASE)))
+    val lowerCharArg = NamedArgument(
+      "lowerChar",
+      FixedArgumentType(StringType),
+      Some(Literal(Mask.MASKED_LOWERCASE)))
+    val digitCharArg = NamedArgument(
+      "digitChar",
+      FixedArgumentType(StringType),
+      Some(Literal(Mask.MASKED_DIGIT)))
+    val otherCharArg = NamedArgument(
+      "otherChar",
+      FixedArgumentType(StringType),
+      Some(Literal(Mask.MASKED_IGNORE)))
+    val functionSignature: FunctionSignature = FunctionSignature(Seq(
+      strArg, upperCharArg, lowerCharArg, digitCharArg, otherCharArg))
+    Seq(functionSignature)
+  }
+
+  override def build(funcName: String, expressions: Seq[Expression]): Expression = {
+    if (expressions.length < 1 || expressions.length > 5) {
+      throw QueryCompilationErrors.wrongNumArgsError(
+        funcName, Seq(1, 2, 3, 4, 5), expressions.length)
+    }
+    new Mask(expressions(0), expressions(1), expressions(2), expressions(3), expressions(4))
```

Comment:
here it seems like by this point, since we provided an override for `functionSignatures`, that we have the guarantee that the `expressions` that show up here are all 5 expressions from the function signature, even if one or more of the provided arguments are missing (and we substitute the corresponding default expressions instead). Can you comment this on the `build` method inside the base `ExpressionBuilder` to make sure the semantics are clear?